File: game/IdleMarket/Assets/Scripts/Equipments/Equipment.cs
using System.Collections.Generic;

[System.Serializable]
public class Equipment
{
    public string id;
    public string userId;

    public EquipmentType equipmentType;
    public int rarity;
    public StatType mainStat;
    public float mainStatValue;
    public int rating;

    public List<SubStat> subStats = new();

    public bool isEquipped;
    public bool isForSale;
    public int? salePrice;

    public Equipment(EquipmentType equipmentType, int rarity, StatType mainStat, float mainStatValue, int rating, List<SubStat> subStats)
    {
        this.equipmentType = equipmentType;
        this.rarity = rarity;
        this.mainStat = mainStat;
        this.mainStatValue = mainStatValue;
        this.rating = rating;
        this.subStats = subStats;
    }
}

public class SubStat
{
    public StatType statType;
    public float statValue;

    public SubStat(StatType statType, float statValue)
    {
        this.statType = statType;
        this.statValue = statValue;
    }
}

File: web/src/types/equipment.ts
export type StatType =
  | "Health"
  | "Attack"
  | "Defense"
  | "Speed"
  | "CriticalChance"
  | "CriticalDamage";

export type EquipmentType = "Helmet" | "Armor" | "Boots" | "Sword";

export interface SubStat {
  id: string;
  statType: StatType;
  statValue: number;
}

export interface Equipment {
  id: string;
  equipmentType: EquipmentType;
  rarity: number;          
  mainStat: StatType;
  mainStatValue: number;
  rating: number;          
  subStats: SubStat[];
  isEquipped: boolean;
  isForSale: boolean;
  salePrice: number | null;
}